azkaban-web-server/src/web/js/azkaban/view/svg-graph.js

```diff
@@ -439,6 +441,50 @@ azkaban.SvgGraphView = Backbone.View.extend({
     this.graphBounds = bounds;
   },
 
+  expandAllFlows: function (node) {
+    if (node) {
+      // expands all embedded flows inside given node
+      if (node.type == 'flow') {
+        this.expandFlow(node);
+      }
+
+      if (node.nodes && node.nodes.length > 0) {
+        for (var i = 0; i < node.nodes.length; ++i) {
+          this.expandAllFlows(node.nodes[i]);
+        }
+      }
+    } else {
+      // expands all embedded flows in the graph
+      var nodes = this.model.get("data").nodes;
+      for (var i = 0; i < nodes.length; ++i) {
+        this.expandAllFlows(nodes[i]);
+      }
+    }
+  },
+
+  collapseAllFlows: function (node) {
+    if (node) {
+      // collapse all embedded flows inside given node
+
+      // collapse already rendered nodes of type flow
+      if (node.type == 'flow' && node.gNode) {
```

Comment:
Why `node.gNode `is checked in `collapseAllFlows` but not `expandAllFlows`?

Reply:
In expandAllFlows is not needed because this method actually triggers the creation of gNode, which is a JavaScript object containing info about the html tag that represents the node visually. This gNode is accessed in collapseAllFlows to hide the element so it needs to exist at this point. 

If we remove this check we get an error whenever we try to collapse before ever expanded because we iterate over all nodes when doing collapsing but some of them, the embedded ones, don’t have a gNode yet.